addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -2077,6 +2074,41 @@ const Wysiwyg = Widget.extend({
 
         return finalOptions;
     },
+    _getBannerCommand: function (title, alertClass, iconClass, description, priority) {
+        return {
+            category: _t('Banner'),
+            name: title,
+            priority: priority,
+            description: description,
+            fontawesome: iconClass,
+            isDisabled: () => this._filterCommandInBanner() || !this.odooEditor.isSelectionInBlockRoot(),
+            callback: () => {
+                const bannerHTML = $(QWeb.render('web_editor.banner', {
+                    label: title,
+                    alertClass: alertClass,
+                    iconClass: iconClass
+                }))[0];
+                this.odooEditor.execCommand('insert', bannerHTML);
+                setSelection(bannerHTML.querySelector('.o_editor_banner > div'), 0);
+            },
+        }
+    },
+    /**
+     * Check if the selection starts inside a banner. This function can be used
+     * as the `isDisabled` property of a command of the PowerBox to disable
+     * a command in banners.
+     * @returns {boolean} true if the command should be filtered
+     */
+    _filterCommandInBanner: function () {
```

Comment:
We should probably write an editor util instead of this as a `wysiwyg` function, that would accept a javascript selector as an argument.

reasoning:
- we already have 2 functions that do the same thing in Knowledge (and that do not protect against a selection without `anchorNode` by the way :grimacing: )
- it does not use `this`, so no reason to be in the wysiwyg
- there are now literally 3 times the same code in Odoo, with only the selector changing :)

We could also use the `closestElement` util instead of the generic `Element.closest` so that the search is limited to the editor.

Reply:
well closest element is less performant (because does the closest and then, does an additional check for the editor) but still safer (as we want to be limited to the editor).